### PR TITLE
Remove the `unmarked_api` feature

### LINKED ISSUE #37981: Remove the `unmarked_api` feature.
According to #22820 and @petrochenkov in https://github.com/rust-lang/rust/pull/37676#discussion_r87702991, this feature-gate should not be used.
It appears that earlier error checking in the compiler prevents it from ever mattering anyway.

### PR DESCRIPTION
**Pull request by jooert**
Closes #37981.

### CODE REVIEW

File: src/librustc/middle/stability.rs [hunk 1]
```diff
@@ -513,23 +513,8 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
...
-                    self.sess.struct_span_err(span, "use of unmarked library feature")
-                             .span_note(span, "this is either a bug in the library you are \
-                                               using or a bug in the compiler - please \
-                                               report it in both places")
-                             .span_note(span, "use #![feature(unmarked_api)] in the \
-                                               crate attributes to override this")
-                             .emit()
-                }
+                span_bug!(span, "Encountered unmarked API, which should \
+                                 not exist in the standard library");
```
Comment (petrochenkov) [inline src/librustc/middle/stability.rs]: Encountering unmarked API is not a library's  fault, but a compiler bug. Could you tweak the message to not mention the standard library, e.g. just "encountered unmarked API".

**Comment by petrochenkov** [comment]
r=me with tweaked message

**Comment by jooert** [comment]
Sure, I have changed it to what you proposed.

**Pull request merged**